src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -663,14 +690,14 @@ public CreateContainerCmd fillContainerConfig(CreateContainerCmd containerConfig
             containerConfig.withExtraHosts(extraHosts.toArray(new String[extraHosts.size()]));
         }
 
-        if (shmSize != null && shmSize > 0) {
+        if (hostConfig != null && shmSize != null && shmSize > 0) {
             final Long shmSizeInByte = shmSize * 1024L * 1024L;
-            containerConfig.getHostConfig().withShmSize(shmSizeInByte);
+            hostConfig.withShmSize(shmSizeInByte);
         }
 
         final List<String> securityOptions = getSecurityOpts();
-        if (CollectionUtils.isNotEmpty(securityOptions)) {
-            containerConfig.getHostConfig().withSecurityOpts( securityOptions );
+        if (hostConfig != null && CollectionUtils.isNotEmpty(securityOptions)) {
+            hostConfig.withSecurityOpts( securityOptions );
```

Comment:
Please undo these changes; they are unnecessary.
The established pattern in this code (and in the docker-java code) is that "everyone calls getHostConfig() every time they need it" rather than calling it once and reusing it.